csrc/multi_tensor_adam.cu

```diff
@@ -269,7 +386,7 @@ void multi_tensor_adam_cuda(
       chunk_size,
       noop_flag,
       tensor_lists,
-      AdamFunctor<scalar_t_0>(),
+      AdamFunctor<scalar_t_0, float>(),
```

Comment:
could be regarded as backward compatibility breaking but this makes the kernel safer.

Reply:
@crcrpar I can limit my changes to only affect the `use_master=True` case, although I do think the original code was broken in that it doesn't train at all when I pass in FP16 model parameters as it that will try to use FP16 m and v values.

Reply:
yeah I'm comfortable with this functional improvement. appreciate your change :)